db_test.go

```diff
@@ -256,10 +256,6 @@ func Test_DB_WithStorage(t *testing.T) {
 		dynparquet.NewSampleSchema(),
 	)
 
-	// TODO(asubiotto): These buckets are not deleted from disk, resulting in
```

Comment:
Has this been fixed? I realize you can reproduce this easily by adding a panic at the end of the test, indicating that the issue is probably that something is not deferred properly:
```
$ go test -run Test_DB_WithStorage
--- FAIL: Test_DB_WithStorage (0.01s)
    sync.go:69: level=debug db=test msg="syncing block"

    sync.go:69: level=debug db=test msg="done syncing block"

    sync.go:69: level=debug db=test msg="read blocks" n=0

    sync.go:69: level=debug db=test msg="read blocks" n=0

panic: boom [recovered]
	panic: boom

goroutine 20 [running]:
testing.tRunner.func1.2({0x10523abc0, 0x105343188})
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1396 +0x1c8
testing.tRunner.func1()
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1399 +0x378
panic({0x10523abc0, 0x105343188})
	/opt/homebrew/opt/go/libexec/src/runtime/panic.go:884 +0x204
github.com/polarsignals/frostdb.Test_DB_WithStorage(0x140004a0820)
	/Users/asubiotto/Developer/github.com/polarsignals/frostdb/db_test.go:340 +0x800
testing.tRunner(0x140004a0820, 0x105340ce0)
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1446 +0x10c
created by testing.(*T).Run
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1493 +0x300
exit status 2
FAIL	github.com/polarsignals/frostdb	0.365s
$ go test -run Test_DB_WithStorage
--- FAIL: Test_DB_WithStorage (0.01s)
    sync.go:69: level=debug db=test msg="syncing block"

    sync.go:69: level=debug db=test msg="done syncing block"

    sync.go:69: level=debug db=test msg="read blocks" n=1

    sync.go:69: level=debug db=test msg="read blocks" n=1

    db_test.go:336:
        	Error Trace:	db_test.go:336
        	            				physicalplan.go:69
        	            				filter.go:244
        	            				table.go:696
        	            				errgroup.go:75
        	            				asm_arm64.s:1172
        	Error:      	Not equal:
        	            	expected: 2
        	            	actual  : 4
        	Test:       	Test_DB_WithStorage
panic: boom [recovered]
	panic: boom

goroutine 20 [running]:
testing.tRunner.func1.2({0x102882bc0, 0x10298b188})
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1396 +0x1c8
testing.tRunner.func1()
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1399 +0x378
panic({0x102882bc0, 0x10298b188})
	/opt/homebrew/opt/go/libexec/src/runtime/panic.go:884 +0x204
github.com/polarsignals/frostdb.Test_DB_WithStorage(0x140004184e0)
	/Users/asubiotto/Developer/github.com/polarsignals/frostdb/db_test.go:340 +0x800
testing.tRunner(0x140004184e0, 0x102988ce0)
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1446 +0x10c
created by testing.(*T).Run
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1493 +0x300
exit status 2
FAIL	github.com/polarsignals/frostdb	0.202s
```
On the first run, the panic is hit, on the second run the test fails with an error because more rows are read than expected.

Reply:
Did you run this with the fix in this PR? 

```
thor@thors-MacBook-Pro frostdb % go test -v -run WithStorage
=== RUN   Test_DB_WithStorage
    sync.go:69: level=debug db=Test_DB_WithStorage msg="syncing block"

    sync.go:69: level=debug db=Test_DB_WithStorage msg="done syncing block"

    sync.go:69: level=debug db=Test_DB_WithStorage msg="read blocks" n=1

    sync.go:69: level=debug db=Test_DB_WithStorage msg="read blocks" n=1

--- FAIL: Test_DB_WithStorage (0.01s)
panic: boom [recovered]
	panic: boom

goroutine 5 [running]:
testing.tRunner.func1.2({0x1009fef40, 0x100b18620})
	/usr/local/go/src/testing/testing.go:1396 +0x1c8
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1399 +0x378
panic({0x1009fef40, 0x100b18620})
	/usr/local/go/src/runtime/panic.go:884 +0x204
github.com/polarsignals/frostdb.Test_DB_WithStorage(0x1400029d520)
	/Users/thor/go/src/github.com/polarsignals/frostdb/db_test.go:336 +0x7e4
testing.tRunner(0x1400029d520, 0x100b16118)
	/usr/local/go/src/testing/testing.go:1446 +0x10c
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1493 +0x300
exit status 2
FAIL	github.com/polarsignals/frostdb	0.219s
thor@thors-MacBook-Pro frostdb % go test -v -run WithStorage
=== RUN   Test_DB_WithStorage
    sync.go:69: level=debug db=Test_DB_WithStorage msg="syncing block"

    sync.go:69: level=debug db=Test_DB_WithStorage msg="done syncing block"

    sync.go:69: level=debug db=Test_DB_WithStorage msg="read blocks" n=1

    sync.go:69: level=debug db=Test_DB_WithStorage msg="read blocks" n=1

--- FAIL: Test_DB_WithStorage (0.01s)
panic: boom [recovered]
	panic: boom
```
	
I'm unable to reproduce this on this branch.

Reply:
Fixed! Thanks